components/automate-ui/src/app/components/create-object-modal/create-object-modal.component.ts

```diff
@@ -76,7 +77,9 @@ export class CreateObjectModalComponent implements OnInit, OnDestroy, OnChanges
       this.policiesUpdatedEvent.emit();
 
       if (this.createProjectModal) {
```

Comment:
Regarding special casing in this code, here's a thought that makes it slightly less of a violation of the Open/Closed Principle: don't check who the caller is, rather just let the caller specify a need:
```suggestion
      if (this.associateTeamsAndPolicies) {
```